### DISCUSSION TITLE
Ignore un-fetchable optionalDependencies

### DISCUSSION THREAD

**Discussion by cprecioso** [yarnpkg/berry#2780] [Q&A] [answered]
Hi! I have a question, I'm consuming a package that has some of their dependencies as `optionalDependencies`. So far so good.
However, this optional dependencies **can't be fetched** at all, they're private. The library can deal with this, and gracefully degrades when these optional dependencies are not present.
Installing with yarn@2.4.1, I receive a HTTP 404 error, and the install does not complete.
This behaviour is explained in #2425. However, I don't have access to the original package, I'm just a consumer of it. Is there any workaround for yarn to treat not finding these dependencies as a skippable error?
Thanks!

**Comment by merceyz** [comment, marked as answer]
> I'm consuming a package that has some of their dependencies as optionalDependencies. So far so good.
However, this optional dependencies can't be fetched at all, they're private

`optionalDependencies` is dependencies that doesn't have to have a successful build, however they can't be missing, so that's something the publisher of that package should change. (https://github.com/yarnpkg/berry/issues/2425#issuecomment-770361283)

> However, I don't have access to the original package, I'm just a consumer of it. Is there any workaround for yarn to treat not finding these dependencies as a skippable error?

You can use [resolutions](https://yarnpkg.com/configuration/manifest#resolutions) to override them to a empty package.

**Reply by alecmev** [reply]
For future readers, there's no special "empty package", just need to resolve to a noop package available on npm like `npm:empty-npm-package@1.0.0` (found in [another discussion](https://github.com/yarnpkg/berry/discussions/4794)).
